backend/src/app.ts

```diff
@@ -57,6 +57,34 @@ app.post(
   }
 );
 
+/**
+ * User sign up/sign in
+ */
+app.post(
+  '/userLoginSignup',
+  [middleware.authenticateUser, middleware.getDB],
+  async (req: express.Request, res: express.Response) => {
+    let user = await getUser(req.db, req.user.oid);
+
+    // create if not found
+    if (!user) {
+      const newUser: User = {
+        primaryKey: `USER#${req.user.oid}`,
+        sortKey: `METADATA#${req.user.oid}`,
+        faculty: undefined,
+        imInto: undefined,
```

Comment:
The database we are using is called a NoSQL database (as opposed to a SQL database, you may also see it called a relational database). With a relational database, it is more strict - as you've done here, we need to set all the other columns to undefined. But the cool thing about NoSQL is that we don't need to do that. 

We can create a new user that ONLY has a primaryKey and sortKey. The next endpoint can add the faculty, what the user is into, and anything else their interested in. Hypothetically, the next endpoint could add a new column called `randomColumnNameThatIsNeverUsed` = 'something' and it will still be stored in a NoSQL DB.

This is a huge benefit to us because who knows what other data we'd want to store with the user in the future (e.g. maybe we will want to ask them what year they started at UNSW).

So here, rather than setting values to undefined, let's just not set them at all and let the next endpoint handle it.